R/eval-relocate.R

```diff
@@ -67,12 +67,18 @@ eval_relocate <- function(expr,
                           name_spec = NULL,
                           allow_rename = TRUE,
                           allow_empty = TRUE,
+                          allow_predicates = NA,
                           before_arg = "before",
                           after_arg = "after",
                           env = caller_env(),
                           error_call = caller_env()) {
   check_dots_empty()
 
+  if (is.na(allow_predicates)) {
+    allow_predicates <- tidyselect_data_supports_predicates(data)
```

Comment:
If the frontend doesn't support predicates and `eval_select(allow_predicates = TRUE)` has been invoked, should it be:

- An error?
- The frontend has the last word?

I think it would make sense for the caller of `eval_select()` to turn off predicate support, but not enable it if that isn't possible.

Reply:
Oh yeah, it probably should be `allow_predicates <- allow_predicates && tidyselect_data_supports_predicates(data)`